### ISSUE TITLE
[Bug] Build assumes repo root is at ~/libra

### ISSUE THREAD

**Issue by dboreham** [OLSF/libra#433]
Breaks the build if you don't/can't put your tree there:

```
   Compiling miner v0.1.0 (/workspaces/libra/miner)
    Finished release [optimized + debuginfo] target(s) in 8m 40s
sudo cp -f /home/vscode/libra/target/release/miner /usr/local/bin/miner
cp: cannot stat '/home/vscode/libra/target/release/miner': No such file or directory
```

**Comment by dboreham** [comment]
https://github.com/OLSF/libra/blob/v4/Makefile#L9

**Comment by dboreham** [comment]
Fix:

```
diff --git a/Makefile b/Makefile
index 4504db7f3..2d2a01996 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,9 @@ DATA_PATH = ${HOME}/.0L
 CHAIN_ID = 1

 ifndef SOURCE
-SOURCE=${HOME}/libra
+MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
+MAKEFILE_DIR := $(dir $(MAKEFILE_PATH))
+SOURCE=${MAKEFILE_DIR}
 endif

 ifndef V
```

**Comment by lpgeiger** [comment]
closed with release-v4.3.0
